test/cases/test_clipboard.c

```diff
@@ -0,0 +1,64 @@
+#include <stdlib.h>
+#include <LCUI.h>
+#include <LCUI/gui/widget.h>
+
+#include <LCUI/gui/widget/textedit.h>
+#include <LCUI/timer.h>
+#include <LCUI/input.h>
+#include "ctest.h"
+
+static void CopyText(void *arg)
+{
+	LCUI_Widget w = arg;
+	LCUIWidget_SetFocus(w);
+	LCUI_SysEventRec ev;
+
+	ev.type = LCUI_KEYDOWN;
+	ev.key.code = (int)LCUI_KEY_V;
+	ev.key.ctrl_key = TRUE;
+	ev.key.shift_key = FALSE;
+	LCUI_TriggerEvent(&ev, NULL);
+}
+
+static void OnText1Focused(void *arg)
+{
+	LCUI_Widget w = arg;
+	LCUI_SysEventRec ev;
+	ev.type = LCUI_KEYDOWN;
+	ev.key.code = (int)LCUI_KEY_C;
+	ev.key.ctrl_key = TRUE;
+	ev.key.shift_key = FALSE;
+
+	LCUI_TriggerEvent(&ev, NULL);
+}
+
+static void OnCheckText(LCUI_Widget w, LCUI_WidgetEvent e, void *arg)
```

Comment:
```diff
+ #include <LCUI/clipboard.h>
```
```c
static void OnCheckText(LCUI_Widget w, LCUI_WidgetEvent e, void *arg)
{
	LCUI_Clipboard clipboard = arg;
	it_b("check the pasted text",
	     clipboard != NULL && wcscmp(clipboard->text, L"helloworld") == 0, TRUE);
	LCUI_Quit();
}
```

Reply:
That still won't work right now, since just a NULL is passed without any text. How about, I create a storage for clipboard within `linux_clipboard.c` ? Since it can act as a dummy way to test things.

Also, nice to see you back @lc-soft